libbeat/cfgfile/cfgfile.go

```diff
@@ -62,3 +66,27 @@ func Read(out interface{}, path string) error {
 func IsTestConfig() bool {
 	return *testConfig
 }
+
+// expandEnv replaces ${var} or $var in config according to the values of the
+// current environment variables. The replacement is case-sensitive. References
+// to undefined variables are replaced by the empty string. A default value
+// can be given by using the form ${var:default value}.
+func expandEnv(config []byte) []byte {
```

Comment:
`os.ExpandEnv(key)` is `os.Expand(key, os.Getenv)` and that's what this does, but with two differences. 1) It logs each replacement which is very important for end-user troubleshooting. 2) It supports setting a default value if you use the form `${var:default value}`.